numpyro/infer/util.py

```diff
@@ -935,6 +942,27 @@ def __call__(self, rng_key, *args, **kwargs):
             model_kwargs=kwargs,
         )
 
+    def __call__(self, rng_key, *args, **kwargs):
+        if self.batch_ndims == 0 or self.params == {} or self.guide is None:
+            return self._call_with_params(rng_key, self.params, args, kwargs)
+        elif self.batch_ndims == 1:
+            param_batch_shape = None
+            for param in self.params.values():
+                if param_batch_shape is None:
+                    assert param.shape[: self.batch_ndims]
+                    param_batch_shape = param.shape[: self.batch_ndims]
+                assert param.shape[: self.batch_ndims] == param_batch_shape
```

Comment:
In numpyro, we allow `param` to have arbitrary pytree structure (e.g. the ones in flax/haiku modules) so `param.shape` will not be available. I think you can do
```
batch_size = jnp.shape(tree_flatten(self.params)[0][0])[0]
rng_keys = random.split(rng_key, batch_size)
```

Reply:
Makes sense. I'll change it.